language-plutus-core/plutus-ir/Language/PlutusIR/Analysis/Dependencies.hs

```diff
@@ -31,6 +40,13 @@ data Node = Variable PLC.Unique | Root deriving (Show, Eq, Ord)
 -- vertices are 'Node's.
 type DepGraph g = (G.Graph g, (G.Vertex g)~Node)
 
+varStrictnessFun ::
+    (MonadState DepState m, PLC.HasUnique name u)
```

Comment:
Yes, we rely on global uniqueness. This pass already did, and we run the renamer before the PIR compilation pipeline starts.

Reply:
Well, this pass does rely on global uniqueness but apparently we don't run the renamer before it, so I don't know how everything is not broken.

Reply:
Yes we do, panic over. We just do it inside. That's a bit subtle. I'm unsure whether given a pipeline `passA >=> passB` where B requires global uniqueness we should put the renaming inside `B` or do `passA >=> rename >=> passB`...

Reply:
Normally, the passes should be self-contained, because you may want to reorder them later and because it just generally makes sense to fulfill an assumption yourself rather than rely on the caller. So I think doing renaming inside is perfectly sensible.